### PR TITLE
[CP-1734] Clean up developers.strava.com documentation

### PR DESCRIPTION
**Pull request by zackisaacs**
I've been taking some notes as I went through my first pull request on developers.strava.com, and I just want to update the documentation to reflect some chokepoints and discrepancies.

Completes [CP-1734](https://strava.atlassian.net/browse/CP-1734)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,9 +29,9 @@ We rely on [Docker](https://www.docker.com/) to build and run the site. Download
 
 The OpenAPI specification maintained in this repository, along with the various models it refers to, are not checked-in in their final form. In some cases, such as in production, we want to publish a specification where all files use absolute `$ref` values. In other cases, such as when you want to work on the spec locally, files may use relative references.
 
-If you seek to make local changes to the the API specification you will first need to generate it all. Templates are defined in [Mustache](https://mustache.github.io/), which you may need to install first. On macOS, you may use Homebrew:
+If you seek to make local changes to the the API specification you will first need to generate it all. Templates are defined in [Mustache](https://mustache.github.io/), which you may need to install first. On macOS, you may use [npm](https://www.npmjs.com/):
```
Comment (jeffremer) [inline README.md]: Not everyone always has Node.js/NPM - `mustache` is also a `ruby` gem - `gem install mustache` works as well.

Comment (zackisaacs) [reply README.md]: Thanks, I'll communicate both options here.

File: README.md [hunk 2]
```diff
@@ -29,9 +29,13 @@ We rely on [Docker](https://www.docker.com/) to build and run the site. Download
...
 
-If you seek to make local changes to the the API specification you will first need to generate it all. Templates are defined in [Mustache](https://mustache.github.io/), which you may need to install first. On macOS, you may use Homebrew:
+If you seek to make local changes to the the API specification you will first need to generate it all. Templates are defined in [Mustache](https://mustache.github.io/), which you may need to install first. On macOS, you may install mustache as a gem or by using [npm](https://www.npmjs.com/). As a gem:
 
-    $ brew install mustache
+    $ gem install mustache
+    
+Alternatively, with npm:
+
+    $ npm install mustache --save
```
Comment (jsilland) [inline README.md]: `npm` is not bundled with most OSes, but `gem` is — I would prefer if we avoided offering multiple ways to achieve the same and instead focused on a canonical supported way

File: README.md [hunk 3]
```diff
@@ -67,7 +71,7 @@ After that, you need to run the `io.swagger.codegen.Codegen` class and make sure
 
     $ java -cp codegen/target/static-html-codegen-1.0.0.jar:$(brew --prefix swagger-codegen)/libexec/swagger-codegen-cli.jar io.swagger.codegen.Codegen -i static/swagger/swagger.json -c config/strava-html.json -l strava-html -o content/docs
 
-This will generate the documentation in the location expected by the site.
+This will generate the documentation in the location expected by the site. If the documentation compiles successfully for master but not for your local branch, make sure that you're escaping all [JSON-restricted characters](http://www.json.org/).
```
Comment (jsilland) [inline README.md]: This feels a little overly specific but without providing details — if you want to document the gotchas of the system, I would recommend doing so in a "Troubleshooting" subsection that succinctly explains both the cause and the remedy

File: README.md (hunk 2 above)
Comment (zackisaacs) [reply README.md]: Good point about multiple ways being unnecessary -- I'll stick with the gem installation.

I think it's worth noting that web engineers are instructed to install npm in the Getting Started docs to manage active assets, so using npm shouldn't be extra work for most people.

File: README.md (hunk 3 above)
Comment (zackisaacs) [reply README.md]: Ok, good point. I'll create a "Troubleshooting" subsection.

File: README.md [hunk 4]
```diff
@@ -69,6 +69,10 @@ After that, you need to run the `io.swagger.codegen.Codegen` class and make sure
 
 This will generate the documentation in the location expected by the site.
 
+### Troubleshooting
```
Comment (zackisaacs) [inline README.md]: @jsilland PTAL. Here's the new "Troubleshooting" subsection.

**Comment by zackisaacs** [comment]
Friendly ping @jsilland.

**Comment by zackisaacs** [comment]
pingping @jsilland

**Comment by zackisaacs** [comment]
@jsilland pingpingping!¡ This one is a super easy review.

**Pull request merged**